VideoLocker/src/main/java/org/edx/mobile/view/CourseHandoutFragment.java

```diff
@@ -1,68 +1,78 @@
 package org.edx.mobile.view;
 
 import android.annotation.SuppressLint;
+import android.content.Context;
 import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.support.annotation.StringRes;
 import android.text.TextUtils;
 import android.util.Xml.Encoding;
 import android.view.LayoutInflater;
 import android.view.View;
 import android.view.ViewGroup;
 import android.webkit.WebView;
-import android.widget.ProgressBar;
+import android.widget.TextView;
 
 import com.google.inject.Inject;
+import com.joanzapata.iconify.Icon;
+import com.joanzapata.iconify.IconDrawable;
+import com.joanzapata.iconify.fonts.FontAwesomeIcons;
 
 import org.edx.mobile.R;
+import org.edx.mobile.base.BaseFragment;
 import org.edx.mobile.core.IEdxEnvironment;
+import org.edx.mobile.event.NetworkConnectivityChangeEvent;
 import org.edx.mobile.logger.Logger;
 import org.edx.mobile.model.api.EnrolledCoursesResponse;
 import org.edx.mobile.model.api.HandoutModel;
 import org.edx.mobile.module.analytics.ISegment;
 import org.edx.mobile.task.GetHandoutTask;
 import org.edx.mobile.util.NetworkUtil;
-import org.edx.mobile.util.UiUtil;
 import org.edx.mobile.util.WebViewUtil;
 import org.edx.mobile.view.custom.URLInterceptorWebViewClient;
 
-import org.edx.mobile.base.BaseFragment;
+import de.greenrobot.event.EventBus;
+import roboguice.inject.InjectExtra;
+import roboguice.inject.InjectView;
 
 public class CourseHandoutFragment extends BaseFragment {
-
-    public WebView webview;
-
-    static public String TAG = CourseHandoutFragment.class.getCanonicalName();
-    static public String ENROLLMENT = TAG + ".enrollment";
     protected final Logger logger = new Logger(getClass().getName());
 
+    @InjectExtra(Router.EXTRA_ENROLLMENT)
+    private EnrolledCoursesResponse courseData;
+
     @Inject
     ISegment segIO;
 
     @Inject
     IEdxEnvironment environment;
 
-    @Override
-    public void onCreate(Bundle savedInstanceState) {
-        super.onCreate(savedInstanceState);
+    @InjectView(R.id.webview)
+    private WebView webview;
 
-        final Bundle bundle = getArguments();
-        EnrolledCoursesResponse courseData = (EnrolledCoursesResponse) bundle
-                .getSerializable(ENROLLMENT);
+    @InjectView(R.id.no_coursehandout_tv)
+    private TextView errorTextView;
 
+    private boolean isHandoutFetched;
 
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        EventBus.getDefault().register(this);
```

Comment:
The event bus should be registered on `onViewCreated()`, and unregistered in `onDestroyView()` since we only interact with the views in the callback.